controls/models.py

```diff
@@ -326,9 +326,23 @@ def assign_baseline_controls(self, user, baselines_key, baseline_name):
 
     def statements(self, statement_type):
         """Return on the statements of statement_type produced by this element"""
+
         smts = Statement.objects.filter(producer_element = self, statement_type = statement_type)
         return smts
 
+    def consuming_systems(self):
+        """Return list of systems for which Element is producer_element of statement of type control_implementation"""
+
+        root_element_ids = set([ce['consumer_element'] for ce in Statement.objects.filter(producer_element=self).values('consumer_element').distinct()])
+        systems = [Element.objects.get(pk=root_element_id).system.all()[0] for root_element_id in root_element_ids if root_element_id]
+        # Remove orphaned systems (e.g., systems whose projects have been deleted). See issue https://github.com/GovReady/govready-q/issues/1617
+        systems_with_projects = []
+        for s in systems:
+            if len(s.projects.all()) > 0:
```

Comment:
Change:
if s.projects.exists()

Reply:
Changed.

Reply:
Resolved in 380674c26daf3f29980c27b4480c11a091f4176c